src/v/pandaproxy/schema_registry/seq_writer.cc

```diff
@@ -115,6 +115,15 @@ ss::future<> seq_writer::read_sync() {
     co_await wait_for(max_offset - model::offset{1});
 }
 
+ss::future<> seq_writer::check_mutable(std::optional<subject> const& sub) {
```

Comment:
nit: slightly confusingly named since "mutable" also applies to the mode setting.

Reply:
> nit: slightly confusingly named since "mutable" also applies to the mode setting.

Which mode setting? The `node_config`? - it is that, But is not clear since `force` is somewhat overloaded (from the user, and if applying from the `_schemas` subject.

Reply:
yeah, I was referring to the mode config

>it is that

It is? Aren't we checking readonly mode for actual subject changes?

Reply:
I guess my point is that both `store::check_mutable` and `seq_writer::check_mutable`, but they mean different things, don't they? Not a huge deal; difference should be clear from context.

Reply:
> I guess my point is that both `store::check_mutable` and `seq_writer::check_mutable`, but they mean different things, don't they? Not a huge deal; difference should be clear from context.

To be fair, it's not very clear. I didn't really want to rename `force` to make it clearer.I shall, it is important

Reply:
>Naming suggestion for the first one? `is_canonical`? `always_apply`?

I'd probably lean towards `is_canonical`; tells something about "why"

Reply:
Oh, I misremembered, the `force` confusion is for mode IMPORT, so I'll do that refactoring on that PR.

Reply:
ah, makes sense. Anyway change looks good.